### PR TITLE
Fix rewrap based on stream with an estimate duration

### PR DESCRIPTION
**Pull request by cchampet**

### CODE REVIEW

File: src/AvTranscoder/decoder/AudioDecoder.cpp [hunk 1]
```diff
@@ -97,7 +97,7 @@ bool AudioDecoder::decodeNextFrame(Frame& frameBuffer)
                                         &got_frame, &data.getAVPacket());
         if(ret < 0)
         {
-            throw std::runtime_error("an error occured during audio decoding" + getDescriptionFromErrorCode(ret));
+            throw std::runtime_error("An error occured during audio decoding: " + getDescriptionFromErrorCode(ret));
```
Comment (valnoel) [inline src/AvTranscoder/decoder/AudioDecoder.cpp]: "occu __rr__ ed"

File: src/AvTranscoder/properties/VideoProperties.hpp [hunk 2]
```diff
@@ -40,10 +40,18 @@ class AvExport VideoProperties : public StreamProperties
     Rational getSar() const; // sample/pixel aspect ratio
     Rational getDar() const; // display aspect ratio
 
-    size_t getBitRate() const; ///< in bits/s, 0 if unknown
+    /**
+     * @return The video bitrate in bits/s.
+     * @warning If there is no such info available in the container, this data is estimate by decoding the first GOP.
```
Comment (valnoel) [inline src/AvTranscoder/properties/VideoProperties.hpp]: "estimate __d__"

File: src/AvTranscoder/properties/VideoProperties.hpp [hunk 3]
```diff
@@ -40,10 +40,18 @@ class AvExport VideoProperties : public StreamProperties
...
+     */
+    size_t getBitRate() const;
     size_t getMaxBitRate() const;
     size_t getMinBitRate() const;
-    size_t getNbFrames() const; ///< 0 if unknown
+    /**
+     * @note 0 if unknown.
+     * @warning In case of a raw format, this data is estimate from the fps and the duration.
```
Comment (valnoel) [inline src/AvTranscoder/properties/VideoProperties.hpp]: "estimate __d__"

File: src/AvTranscoder/properties/VideoProperties.hpp [hunk 4]
```diff
@@ -63,7 +71,8 @@ class AvExport VideoProperties : public StreamProperties
 
     /**
      * @brief Override method.
-     * @return the stream duration in seconds, 0 if not available
+     * @return The stream duration in seconds, 0 if not available.
+     * @warning In case of a raw format, this data is estimate from the file size.
```
Comment (valnoel) [inline src/AvTranscoder/properties/VideoProperties.hpp]: "estimate __d__"

File: src/AvTranscoder/decoder/VideoDecoder.cpp [hunk 5]
```diff
@@ -95,7 +95,7 @@ bool VideoDecoder::decodeNextFrame(Frame& frameBuffer)
                                               &got_frame, &data.getAVPacket());
         if(ret < 0)
         {
-            throw std::runtime_error("an error occured during video decoding - " + getDescriptionFromErrorCode(ret));
+            throw std::runtime_error("An error occured during video decoding: " + getDescriptionFromErrorCode(ret));
```
Comment (valnoel) [inline src/AvTranscoder/decoder/VideoDecoder.cpp]: "occu __rr__ ed"

Comment (cchampet) [inline src/AvTranscoder/decoder/VideoDecoder.cpp]: Fixed here cb2d7c4

File: src/AvTranscoder/properties/VideoProperties.hpp (hunk 4 above)
Comment (cchampet) [inline src/AvTranscoder/properties/VideoProperties.hpp]: Fixed here cb2d7c4

File: src/AvTranscoder/properties/VideoProperties.hpp (hunk 3 above)
Comment (cchampet) [inline src/AvTranscoder/properties/VideoProperties.hpp]: Fixed here cb2d7c4

File: src/AvTranscoder/properties/VideoProperties.hpp (hunk 2 above)
Comment (cchampet) [inline src/AvTranscoder/properties/VideoProperties.hpp]: Fixed here cb2d7c4

File: src/AvTranscoder/decoder/AudioDecoder.cpp (hunk 1 above)
Comment (cchampet) [inline src/AvTranscoder/decoder/AudioDecoder.cpp]: Fixed here cb2d7c4

**Comment by cchampet** [comment]
See PR #263

**Pull request closed without merging**